models/repo_tree.go

```diff
@@ -0,0 +1,79 @@
+// TODO:
+// With model objects here, we can write tests for them and for TreeListing? Or maybe
+// they should go elsewhere?
+
+// Then we can write fuller integration tests, with model objects that we're expecting and asserting
+// against
+import (
+	"path/filepath"
+
+	"code.gitea.io/git"
+)
+
+// RepoFile represents a file blob contained in the repository
+type RepoFile struct {
+	Path     string            `json:"path"`
+	// Mode     git.EntryMode     `json:"mode"`  // TODO: Do we include this? It'll require exporting the mode as public in the `git` module...
+	Type     git.ObjectType    `json:"type"`
+	// Size     int64             `json:"size"` // TODO: Do we include this? It's expensive...
```

Comment:
It's not though, since `git cat-file` prints the size along with the ref and filename :/